qualtran/surface_code/algorithm_summary.py

```diff
@@ -41,3 +41,21 @@ class AlgorithmSummary:
     toffoli_gates: float = _PRETTY_FLOAT
     rotation_gates: float = _PRETTY_FLOAT
     rotation_circuit_depth: float = _PRETTY_FLOAT
+
+    def __mul__(self, other: int) -> 'AlgorithmSummary':
+        if not isinstance(other, int):
+            raise TypeError(
+                f'Multiplication isn\'t supported between AlgorithmSummary and non integer type {type(other)}'
```

Comment:
extremely minor nit: Usually we'd just switch to `"`-strings instead of escaping the apostrophe